pool/pool.go

```diff
@@ -63,20 +74,40 @@ func DefaultPollingParams() *ContainerPollingParams {
 }
 
 // AddNode adds address/weight pair to node PoolBuilder list.
-func (pb *Builder) AddNode(address string, weight float64) *Builder {
-	pb.addresses = append(pb.addresses, address)
-	pb.weights = append(pb.weights, weight)
+func (pb *Builder) AddNode(address string, priority int, weight float64) *Builder {
+	pb.nodeParams = append(pb.nodeParams, NodeParam{
+		address:  address,
+		priority: priority,
+		weight:   weight,
+	})
 	return pb
 }
 
 // Build creates new pool based on current PoolBuilder state and options.
 func (pb *Builder) Build(ctx context.Context, options *BuilderOptions) (Pool, error) {
-	if len(pb.addresses) == 0 {
+	if len(pb.nodeParams) == 0 {
 		return nil, errors.New("no NeoFS peers configured")
 	}
 
-	options.weights = adjustWeights(pb.weights)
-	options.addresses = pb.addresses
+	nodeParams := make(map[int]*NodesParam)
+	for _, param := range pb.nodeParams {
+		nodes, ok := nodeParams[param.priority]
+		if !ok {
+			nodes = &NodesParam{priority: param.priority}
+		}
+		nodes.addresses = append(nodes.addresses, param.address)
+		nodes.weights = append(nodes.weights, param.weight)
+		nodeParams[param.priority] = nodes
```

Comment:
The line seems odd if we already have a value in `NodeParams`. Maybe we can move it into the block above:
```
if !ok {
			nodes = &NodesParam{priority: param.priority}
			nodeParams[param.priority] = nodes
		}
```

Reply:
We can but I think it may be a little confused (to understand  why we put some struct into the map and after that change this struct we should always remember this struct are stored by pointer). So I think an explicit assignment is better even though it isn't necessary.